### ISSUE TITLE
Correct playback of YouTube videos in 8k resolution on Pico 4

### ISSUE THREAD

**Issue by MikhailGribov** [Igalia/wolvic#499]
I want to watch youtube 8k 360 degree video. Now if you choose 8k quality, the image freezes on one picture and the sound continues to go. Is there any chance to fix this, maybe increase the buffer or something like that?

**Comment by svillar** [comment]
> I want to watch youtube 8k 360 degree video. Now if you choose 8k quality, the image freezes on one picture and the sound continues to go. Is there any chance to fix this, maybe increase the buffer or something like that?

This was already addressed in issue #64. Long story short, it [was verified](https://bugzilla.mozilla.org/show_bug.cgi?id=1634189) that hardware had limitations decoding this kind of >4k streams. That said, that report is from 2019 so things might have very well changed. We'll check that.

**Comment by svillar** [comment]
I think the issue here is that we haven't implemented CPU/GPU throttling with the OpenXR backend yet. I've just tried with the patch that enables that (see #503) for cases like immersive videos and I was able to watch AirPano 8k 360 videos without any problem. 

Note that when negotiating qualities with YouTube we cap the maximum at 4k resolution as above that the playback started to become quite choppy. We should reevaluate that with all these new hardware we support.
